pkg/controller/proxy/controller_test.go

```diff
@@ -0,0 +1,168 @@
+package proxy
+
+import (
+	"context"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/fake"
+	core "k8s.io/client-go/testing"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+
+	"github.com/ianlewis/controllerutil/informers"
+	"github.com/ianlewis/controllerutil/logging"
+
+	"github.com/ianlewis/memcached-operator/pkg/apis/ianlewis.org/v1alpha1"
+	ianlewisorgfake "github.com/ianlewis/memcached-operator/pkg/client/clientset/versioned/fake"
+	ianlewisorginformers "github.com/ianlewis/memcached-operator/pkg/client/informers/externalversions/ianlewis/v1alpha1"
+)
+
+func getKey(o interface{}, t *testing.T) string {
+	if key, err := KeyFunc(o); err != nil {
+		t.Errorf("Unexpected error getting key for %v: %v", o, err)
+		return ""
+	} else {
+		return key
+	}
+}
+
+func alwaysSynced() bool {
+	return true
+}
+
+type fixture struct {
+	t *testing.T
+
+	client            *fake.Clientset
+	ianlewisorgClient *ianlewisorgfake.Clientset
+
+	proxies []*v1alpha1.MemcachedProxy
+
+	// Actions expected to happen on the client.
+	actions []core.Action
+}
+
+func newFixture(t *testing.T) *fixture {
+	return &fixture{
+		t: t,
+	}
+}
+
+func (f *fixture) newController() (*Controller, informers.SharedInformers) {
+	objects := make([]runtime.Object, len(f.proxies))
+	for i, p := range f.proxies {
+		objects[i] = p
+	}
+
+	f.client = fake.NewSimpleClientset()
+	f.ianlewisorgClient = ianlewisorgfake.NewSimpleClientset(objects...)
+
+	informers := informers.NewSharedInformers()
+	proxyInformer := informers.InformerFor(
+		&v1alpha1.MemcachedProxy{},
+		func() cache.SharedIndexInformer {
+			return ianlewisorginformers.NewMemcachedProxyInformer(
+				f.ianlewisorgClient,
+				corev1.NamespaceAll,
+				0,
+				cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+			)
+		},
+	)
+
+	c := New(
```

Comment:
AFAIK, it would be more idiomatic Go to have this be `NewController` rather than just `New`

Reply:
I've heard that when a package is built around a type then it's redundant to put it in the constructor name when it's in the package name. I'm not married to either idea but I liked this because it's shorter.

Reply:
https://golang.org/doc/effective_go.html#composite_literals
http://www.golangpatterns.info/object-oriented/constructors

But a minor point.